### PR TITLE
Fix typo in error stack trace formatting for clearer output

### PR DESCRIPTION
**Pull request by aboitreaud**
Fix a typo in error stack trace formatting by replacing "\\\n" joins with [`traceback.format_exc()`](https://docs.python.org/3/library/traceback.html#traceback.format_exc). The previous implementation used "\\\n" to join lines, resulting in "\n" printed at the beginning of the lines. Fixing it by joining on "\n" would just  add unnecessary line breaks and extra spacing in the stack trace. Since traceback functions already handle line breaks, this fix ensures proper formatting without redundant joins.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by JDarDagran** [comment]
@kacpermuda could you please comment on that? I can see you made current join in https://github.com/apache/airflow/pull/39813

**Comment by kacpermuda** [comment]
Hey @aboitreaud , thanks for the PR ! Apart from the problem you described with the formatting, you are not passing the exception object anymore to the formatting func. I'm thinking if there possibly is some edge case, where we the last exception raised is **not** the one that we have here in the adapter. Have you considered it here?

File: airflow/providers/openlineage/plugins/adapter.py
Comment (uranusjr) [inline airflow/providers/openlineage/plugins/adapter.py:314]:
```suggestion
                stack_trace = traceback.format_exc(error)
```

I think?

**Review by uranusjr — COMMENTED** [review]
Does this change the error message? If so, can we include a comparison between old and new outputs in this PR?

**Comment by aboitreaud** [comment]
Hi @kacpermuda and @uranusjr and thanks for your comments!
Good catch, and thanks for pointing that out! I hadn’t fully considered the possibility of an edge case where the last exception raised might not be the one we have here in the adapter. To address this, I’m reverting back to using `traceback.format_exception(error)` instead of relying on `traceback.format_exc()`. This ensures we are handling the correct exception in all cases.

However, I don't think that joining on "\n" is necessary, here is of what the stack trace would look like:
- the **old output**, by joining on "\\\n": 
```
Traceback (most recent call last):
\n  File "/Users/adrien.boitreaud/import traceback.py", line 6, in cause_error
    result = 1 / 0
\nZeroDivisionError: division by zero
```
-  by joining on "\n": 
```
Traceback (most recent call last):

  File "/Users/adrien.boitreaud/import traceback.py", line 6, in cause_error
    result = 1 / 0

ZeroDivisionError: division by zero
```
- the **new output**, by joining on "":
```
Traceback (most recent call last):
  File "/Users/adrien.boitreaud/import traceback.py", line 6, in cause_error
    result = 1 / 0
ZeroDivisionError: division by zero
```

**Review by kacpermuda — APPROVED** [review]

**Comment by aboitreaud** [comment]
@uranusjr could I please also get your approval following @Kacper's please ?

**Comment by aboitreaud** [comment]
👊 Gentle bump here @uranusjr @mobuchowski, could I get a code owner review please ?

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**
